File: tcp_udp_experiment/multi_lang_udp_tcp_exp/tcp_client.py
import socket

SERVER_ADDR = "127.0.0.1"
PORT = 8008

skt = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
print("start connection_establish")
connection = skt.connect((SERVER_ADDR, PORT))
print("finish connection_establish")

datasize = 100
data = bytes(range(0,100))

print("send_packet")
skt.sendall(data)

File: tcp_udp_experiment/multi_lang_udp_tcp_exp/tcp_client.c

#include <arpa/inet.h>
#include <netinet/in.h>
#include <sys/socket.h>
#include <unistd.h>
#include <stdio.h>
#include <string.h>

int main(void){
    const char *HOST = "127.0.0.1";
    const int PORT = 8008;

    int sock = socket(AF_INET, SOCK_STREAM, 0);

    struct sockaddr_in addr;
    memset(&addr, 0, sizeof(addr));
    addr.sin_family = AF_INET;
    addr.sin_port = htons(PORT);
    inet_pton(AF_INET, HOST, &addr.sin_addr);

    connect(sock, (struct sockaddr *)&addr, sizeof(addr));

    char data[100];
    memset(data, 0, sizeof(data));
    snprintf(data, sizeof(data), "hello world");
    send(sock, data, sizeof(data), 0);

    close(sock);
    return 0;
}
